### PR TITLE
[Generic Formatter] Get a working flat data structure for bulk upload

### PR DESCRIPTION
**Pull request by tsullivan**
Last (?) part of changes for https://github.com/elastic/kibana/pull/21356

### CODE REVIEW

File: src/server/usage/classes/collector_set.js [hunk 1]
```diff
@@ -94,20 +94,6 @@ export class CollectorSet {
     return Promise.all(fetchPromises);
   }
 
-  bulkFormat(data) {
```
Comment (tsullivan) [inline src/server/usage/classes/collector_set.js:97]: It didn't make a lot of sense to have this in OSS; I moved it to `BulkUploader.toBulkUploadFormat `

File: x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.js [hunk 2]
```diff
@@ -119,41 +121,36 @@ export class BulkUploader {
...
+    // with an 'index' instruction, for bulk upload
+    const flat = Object.keys(typesNested).reduce((accum, type) => {
+      return [
+        ...accum,
         { index: { _type: type } },
-        deepMergeAndGroup[type],
-      ]);
-      return accum;
+        {
+          kibana: this._getKibanaInfoForStats(),
```
Comment (tsullivan) [inline x-pack/plugins/monitoring/server/kibana_monitoring/bulk_uploader.js:146]: I had to make `BulkUploader.toBulkUploadFormat` a non-static method to support `this`

File: x-pack/plugins/monitoring/server/kibana_monitoring/collectors/ops_buffer/ops_buffer.js [hunk 3]
```diff
@@ -27,8 +27,14 @@ export function opsBuffer(server) {
...
     flush() {
+      let cloud; // a property that will be left out of the result if the details are undefined
+      const cloudDetails = cloudDetector.getCloudDetails();
+      if (cloudDetails != null) {
+        cloud = { cloud: cloudDetails };
+      }
+
       return {
-        cloud: cloudDetector.getCloudDetails(),
+        ...cloud,
```
Comment (tsullivan) [inline x-pack/plugins/monitoring/server/kibana_monitoring/collectors/ops_buffer/ops_buffer.js:37]: For checking if payloads are empty, this field needed to be taken out when value is undefined

**Comment by tsullivan** [comment]
I don't love the idea of asking collector objects to provide a `formatForBulkUpload`, but I think that requirement is mainly for these ones, which are a little tricky. For new collector objects to come in future enhancements, we'd want to make sure the defaults get them working without having more `formatForBulkUpload` methods.

**Comment by chrisronline** [comment]
The code looks good to me!

How would you suggest we test to ensure parity against the existing code? Use the `scripts/api_debug` against master, then against this branch and compare?

**Comment by tsullivan** [comment]
To test this, I ran a 6.3.2 instance in a docker container and compared its `.monitoring-kibana-*` data with the data indexed in this branch. I also tested that all the parts of the Kibana Monitoring dashboards work.

**Comment by chrisronline** [comment]
Cool, I'm going to merge this in and work off the other branch. Thanks!

**Pull request merged**